middleware/kubernetes/handler.go

```diff
@@ -35,6 +35,8 @@ func (k Kubernetes) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.M
 			}
 			// Set the zone to this specific request.
 			zone = state.Name()
+		} else {
```

Comment:
If you reverse the == "PTR" above like so it read better I think:
~~~
if state.Type() != "PTR" {
  return middleware.NextOrFailure(k.Name(), k.Next, ctx, w, r)
}
~~~
Then the rest can be dedented?